pkg/scheduler/plugins/elasticquota/core/group_quota_manager.go

```diff
@@ -0,0 +1,1311 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/scheduling/v1alpha1"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	quotav1 "k8s.io/apiserver/pkg/quota/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/koordinator-sh/koordinator/cmd/koord-scheduler/app/options"
+	"github.com/koordinator-sh/koordinator/pkg/util"
+)
+
+const (
+	DumpQuotaInfoCycle = time.Duration(10) * time.Second
+)
+
+type GroupQuotaManager struct {
+	Lock                             sync.RWMutex
+	sysQuota                         v1.ResourceList
+	sysQuotaUsed                     v1.ResourceList
+	totalResourceNoSys               v1.ResourceList
+	totalResourceWithSys             v1.ResourceList
+	resourceKeys                     map[string]struct{}
+	quotaInfoMap                     map[QuotaNameType]*QuotaInfo
```

Comment:
If I understand correctly, quotaInfoMap stores all nodes in an adjacency list, which is convenient for us to find all parent nodes of a node (because a node's Quota update will affect all its parent nodes). quotaTreeWrapperMap in turn stores the child nodes of each node, and recalculates the runtime of each child node when the parent node runtime is updated.

Leave a comment here will be very helpful.